File: StockMarketCharting/StockMarket.AdminAPI/Services/CompanyService.cs
using StockMarket.AdminAPI.Models;
using StockMarket.AdminAPI.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StockMarket.AdminAPI.Services
{
    public class CompanyService : ICompanyService
    {
        /*private CompanyRepository _repo;
        public CompanyService()
        {
            _repo = new CompanyRepository();
        }
        public CompanyService(CompanyRepository repo)
        {
            _repo = repo;
        }*/
        private CompanyRepository _repo = new CompanyRepository();
        public void AddCompany(Company item)
        {
            _repo.AddCompany(item);
        }

        public void DeleteCompany(int id)
        {
            _repo.DeleteCompany(id);
        }

        public void EditCompany(Company item)
        {
            _repo.EditCompany(item);
        }

        public List<Company> GetCompanies()
        {
            return _repo.GetCompanies();
        }

        public Company GetCompanyDetails(int id)
        {
            return _repo.GetCompanyDetails(id);
        }
    }
}

File: StockMarketAngular/src/app/services/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaderResponse } from "@angular/common/http";
import { Observable } from "rxjs";
import { Company } from '../models/company';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {
  path: string = "http://localhost:59534/api/Company/"
  //path: string = "http://localhost:8090/api/Company/";

  constructor(private http: HttpClient) { }

  public GetCompanies(): Observable<Company[]>
  {
    return this.http.get<Company[]>(this.path+"GetCompanies")
  }
  
  public GetCompanyDetails(id:number): Observable<any>
  {
    console.log(id);
    return this.http.get<any>(this.path+'GetCompanyDetails/'+id)
  }

  public AddCompany(item:Company)
  {
    console.log(item);
    return this.http.post(this.path+"AddCompany/",item);
  }
  
  public EditCompany(item:Company):Observable<any>
  {
    console.log(item);
    return this.http.put<any>(this.path+"EditCompany/",item);
  }

  public DeleteCompany(id:number):Observable<any>
  {
    console.log(id);
    return this.http.delete<any>(this.path+'DeleteCompany/'+id)
  }
}
